fvm/Equation/Operator.cpp

```diff
@@ -24,24 +26,48 @@ Operator::~Operator() {}
 
 /**
  * Add an advection term to this operator.
+ * The optional argument addAsEquationTerm is included
+ * to make it possible to add also advection-diffusion
+ * terms as ordinary terms, ie not via an AdvectionDiffusionTerm.
+ * This is needed for example for advection-diffusion terms on a
+ * quantity with more multiples than one (e.g. ions in DREAM), as the
+ * AdvectionDiffusionTerm class is presently not adapted for such terms.
  */
-void Operator::AddTerm(AdvectionTerm *a) {
-    if (adterm == nullptr)
-        adterm = new AdvectionDiffusionTerm(this->grid);
+void Operator::AddTerm(AdvectionTerm *a, bool addAsEquationTerm) {
```

Comment:
Could an equivalent effect be achieved by defining a method ``AddTerm(IonChargedAdvectionDiffusionTerm<AdvectionTerm>*)`` instead of using the new flag  ``addAsEquationTerm``?

Reply:
I think so, yes